extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -28,15 +30,71 @@ class AmpStickyAd extends AMP.BaseElement {
   isLayoutSupported(layout) {
     return layout == Layout.NODISPLAY;
   }
-
-  /** @override */
+  /**
+   * @override
+   * @this {undefined}  // Make linter happy
+   */
   buildCallback() {
     /** @const @private {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+    this.viewport_ = this.getViewport();
+    this.isDisplayed_ = false;
+    this.initialScrollTop_ = this.viewport_.getScrollTop();
```

Comment:
`buildCallback` must not do any measurements. Also, it'd seem that `scrollHeight` can change after it's stored here. What would happen in that case?